src/common/TrackedOp.h

```diff
@@ -386,4 +278,155 @@ class TrackedOp : public boost::intrusive::list_base_hook<> {
 };
 
 
+class OpTracker {
+  struct ShardedTrackingData {
+    mutable Mutex ops_in_flight_lock_sharded;
+    TrackedOp::tracked_op_list_t ops_in_flight_sharded;
+    explicit ShardedTrackingData(string lock_name):
+      ops_in_flight_lock_sharded(lock_name.c_str()) {
+    }
+  };
+
+  friend class OpHistory;
+  std::atomic<int64_t> seq = { 0 };
```

Comment:
Probably we do have a false sharing with the instance of `tiny_vector` here. `alignas(64)` could help but I would love to kill `seq` for the sake of `register_inflight_op`. IIUC `TrackedOp::seq` is just an indicator of `OpTracker`'s shard holding given `TrackedOp`. At least in the OSD case, during `create_request`,  there is quite good alternative: `spg_t`. That way we could establish relationship between shards of `OpTracker` the WQ.

Reply:
Hmm, OSD is fine with making the `TrackedOp::seq` `private`.  `MonOpRequest` dumps it but the doors seems open.